### ISSUE TITLE
VersionPress status, 06/2020

### ISSUE THREAD

**Issue by borekb** [versionpress/versionpress#1481]
_I'm creating this issue to have a place to point people to._

As of today, nobody is actively working on VersionPress and it's been this way for a while, actually – last active development was _before_ WordPress 5.0 / Gutenberg.

We, the original team, are still together and working on WordPress-related products but not VersionPress itself. If you're wondering why, the main reason is that VersionPress still requires several man-years of engineering work to be finished and we simply don't have resources to do that. Spending evenings and weekends on it doesn't make much sense to us either – we'd burn our family time while still not _really_ getting VersionPress closer to being "done".

BTW, it's not just about us – any company or individual can take VersionPress and start developing it. We even know about agencies that are quite close to be willing / able to do that, but it's generally hard to do open source and they usually end up maintaining their own internal fork instead. I completely understand this – generalizing something for _all_ users of WordPress is tough.

So that leaves VersionPress in a Developer Preview state and not recommended for production usage. Is it "dead"? It's not actively developed but the repo, the docs site, the Gitter chat room etc. are all up and running because even in its current state, VersionPress seem to be useful to people and we get quite a bit of demand for it still. But it is what it is – a developer preview without any active development around it. We hope this will change one day but there are no guarantees, of course.

Hope this clarifies the situation & thanks for your understanding!

**Comment by roundrobinjp** [comment]
@borekb 
Thanks. This explanation was a very much required one.
I saw a number of people around me trying to use VersionPress, but finally
failed to utilize it properly. You may like to post this to VersionPress
website and facebook page. That will save people from wasting their time.

However, I wonder how people in the world are doing without any solution
like VersionPress. I feel very uncomfortable without a version control system.
How come people are doing without it?
How do they merge the work of team members where database changes are involved?
I would love to know their workflow.

**Comment by borekb** [comment]
People struggle 😄.

This is relatively hard with any CMS by the way, as database is a necessity for performance reasons and that just isn't friendly to version control. WordPress is relatively well-positioned to have this problem solved as it has a stable DB structure, is itself open source, etc., but it's still a tough challenge.

**Comment by amjad** [comment]
Maybe Automattic can buy you out? You have a great name and a useful plugin.

**Comment by borekb** [comment]
Not going to happen 😄.

**Comment by yusufhm** [comment]
Hey @borekb is there a list of priority issues that we could pick up and work on if we wanted to keep seeing some progress on this and contribute back? I've just discovered this plugin and I think it's a great idea for tracking changes on Wordpress sites.

**Comment by gjohnhazel** [comment]
Wow I just discovered this and would absolutely love a complete product, but I definitely understand not being able to continue production. I hope a company picks this up and, if necessary, monetizes to keep it sustainable! This is exactly the solution I was looking for. Thanks for the hard work you have put into this foundation and proof-of-concept!

**Comment by strarsis** [comment]
Other CMS wouldn't support what VersionPress does either? Like JavaScript based ones, or similar ones?
Has the VersionPress dev team switched to another CMS/stack that offers these features out of the box?

**Comment by willem-aart** [comment]
> Other CMS wouldn't support what VersionPress does either? Like JavaScript based ones, or similar ones?
> Has the VersionPress dev team switched to another CMS/stack that offers these features out of the box?

There are git backed CMSes such as https://www.netlifycms.org/

**Comment by nightillusions** [comment]
> Other CMS wouldn't support what VersionPress does either? Like JavaScript based ones, or similar ones?
> Has the VersionPress dev team switched to another CMS/stack that offers these features out of the box?

Yes, I preffer using Statamic (https://statamic.dev/git-integration) but customer still think WordPress does a great job :(

**Comment by davidvc** [comment]
Hi @borekb. If it's not too much trouble, could you summarize the major challenges and big areas of work around getting this production ready? Is it the database integration? Is it Gutenberg?  Just trying to get a sense of how crazy I'd be to try and take that effort on (I'm suspecting, pretty crazy :) )

Thanks,

David

**Comment by yungifez** [comment]
Damn this is a real shame, i want to use wordpress for a project and im afraid one day all the pages would disappear all of a sudden 😆

**Comment by robertandrews** [comment]
Are there any active forks which have sustained the plugin?

**Comment by OnkelTem** [comment]
**@borekb**
> However, I wonder how people in the world are doing without any solution like VersionPress.

That's exactly what I'm thinking about. I've got a wp site which I was going to put under version control, until I discovered that they haven't developed anything for all these years since the emergence of WP. That's a big shame, I'm really disappointed and I wonder why people are so naive and incompetent, that they continue to pick WP.

Speaking of systems that have this issue almost perfectly addressed, I can name Drupal 8. For years now they have `drush` 's `export-config`/`import-config` commands that put Drupal config under version control.

I've already learned that **wp-cli** tool can export and import `posts` and `postmeta` into XML files and export/import options. But that doesn't look enough.

As for this project, I was also excited to discover it and I understand how difficult it would be to finish it. Honestly, I reckon it was almost impossible. But it was a good try! I express sincere respect to the developers.

**Comment by craftydomops** [comment]
I am attempting to work out how to do modern development witih git on Wordpress, with environments, and there doesn't seem to be anything that covers it all. VersionPress feels like the closest.

How are others doing git with Wordpress?

**Comment by HammyHavoc** [comment]
> I am attempting to work out how to do modern development witih git on Wordpress, with environments, and there doesn't seem to be anything that covers it all. VersionPress feels like the closest.
> 
> How are others doing git with Wordpress?

Did you have any joy with this? I'm looking to sync and edit content via git like https://github.com/mAAdhaTTah/wordpress-github-sync but without suffering the WordPress admin area.

**Comment by mtovmassian** [comment]
Too bad...The promise was great.
I can't imagine the project being abandoned while half of the websites in the world are powered by WordPress.

**Comment by giorginogreg** [comment]
@mtovmassian Are you able to maintain with me a fork of this project, or if the officials mantainers of this proj. wants, to become the official maintainers?

**Comment by giorginogreg** [comment]
> Are there any active forks which have sustained the plugin?

@robertandrews I am creating a fork for supporting current WP version (6.5.5)
